distribution/lib/Standard/Base/0.0.0-dev/src/Random.enso

```diff
@@ -1,99 +1,307 @@
-        range = max - min
-        if range < Java_Integer.MAX_VALUE then min + (self.java_random.nextInt range) else
-            Error.throw (Illegal_Argument.Error "Currently only integer ranges of up to 2^31-1 are supported.")
+       - standard_deviation: The standard deviation of the sampled distribution.
+       - mean: The mean of the sampled distribution.
+
+       > Example
+         Generate an gaussian float.
+
+              import Standard.Base.Random.Random
+
+              f = Random.gaussian
+    gaussian : Float -> Float -> Float
+    gaussian (standard_deviation:Float=1.0) (mean:Float=0.0) = Random.default_instance.gaussian standard_deviation mean
+
+    ## GROUP Random
+       Returns `count` indices sampled from the range 0 (inclusive) to n
+       (exclusive) without replacement.
+       
+       Arguments:
+       - n: The end of the range to sample from (exclusive).
+       - count: The number of samples to take.
+
+       > Example
+         Generate 10 random indices between 0 and 100.
+
+              import Standard.Base.Random.Random
+
+              indices = Random.indices 100 10
+    indices : Integer -> Integer -> Vector Integer
+    indices (n:Integer) (count:Integer=1) = Random.default_instance.indices n count
+
+    ## GROUP Random
+       Return a `Date` within the specified `Date_Range`.
+
+       Arguments:
+       - date_range: The `Date_Range` that specifies the range to pick from. The
+         range is inclusve at the lower bound, and exclusive at the upper bound.
+
+       > Example
+         Generate a random date.
+
+             import Standard.Base.Random.Random
+
+             range = Date_Range.new (Date.new 2023 03 01) (Date.new 2023 10 15)
+             d = Random.date range
+    date : Date_Range -> Date
+    date (date_range:Date_Range) = Random.default_instance.date date_range
+
+    ## GROUP Random
+       Return a `Time_Of_Day` between `start_time` and `end_time` (inclusive).
+
+       Arguments:
+       - start_time: The lower bound of the range to pick from (inclusive).
+       - end_time: The upper bound of the range to pick from (inclusive).
+
+       The generated time is at a granularity of 1 second.
+
+       Note: the time range end is inclusive because it is impossible to specify
+       an end time after the last second of the day.
+
+       > Example
+         Generate a random time.
+
+             import Standard.Base.Random.Random
+
+             start = Time_Of_Day.new 8 12 15
+             end = Time_Of_Day.new 9 40 2
+             t = Random.time start end
+    time : Time_Of_Day -> Time_Of_Day -> Time_Of_Day
+    time (start_time:Time_Of_Day=(Time_Of_Day.new 0 0 0)) (end_time:Time_Of_Day=(Time_Of_Day.new 23 59 59)) =
+        Random.default_instance.time start_time end_time
+
+    ## GROUP Random
+       Return a new UUID.
+
+       ! `uuid` does not use the underlying `java.util.Random` instance, and is
+       therefore not affected by the seed.
```

Comment:
```suggestion
       ! Random Seed
       
         `uuid` does not use the underlying `java.util.Random` instance, and is
         therefore not affected by `set_seed`.
```

I'm not sure how this will look like in the new IDE, but currently I think the `!` and `?` sections need a header.